UserDataProcessingTrigger/index.ts

```diff
@@ -22,57 +29,100 @@ export const ProcessableUserDataDownload = t.intersection([
 
-interface ITaskDescriptor {
-  orchestrator: string;
-  id: ProcessableUserDataDownload["userDataProcessingId"];
-  input: ProcessableUserDataDownload;
-}
-
 export function index(
   context: Context,
   input: unknown
-): Promise<readonly string[]> {
+): Promise<ReadonlyArray<string | void>> {
   const dfClient = df.getClient(context);
-  const tasksDescriptors = CosmosDbDocumentCollection.decode(input)
+  const operations = CosmosDbDocumentCollection.decode(input)
     .getOrElseL(err => {
       throw Error(`${logPrefix}: cannot decode input [${readableReport(err)}]`);
     })
     .reduce(
-      (tasks, maybeProcessable) =>
-        ProcessableUserDataDownload.decode(maybeProcessable).fold(
-          _ => {
-            context.log.warn(
-              `${logPrefix}: skipping document [${JSON.stringify(
-                maybeProcessable
-              )}]`
-            );
-            return tasks;
-          },
-          processable => [
-            ...tasks,
-            {
-              id: processable.userDataProcessingId,
-              input: processable,
-              orchestrator: "UserDataDownloadOrchestrator"
-            }
-          ]
-        ),
-      [] as readonly ITaskDescriptor[]
+      (lazyOperations, processableOrNot) =>
+        t
+          .union([
+            ProcessableUserDataDownload,
+            ProcessableUserDataDelete,
+            ProcessableUserDataDeleteAbort
+          ])
+          .decode(processableOrNot)
+          .chain(processable =>
+            fromNullable(undefined)(
+              ProcessableUserDataDownload.is(processable)
+                ? () =>
+                    dfClient.startNew(
+                      "UserDataDownloadOrchestrator",
+                      makeDownloadOrchestratorId(processable.fiscalCode),
+                      processable
+                    )
+                : ProcessableUserDataDelete.is(processable)
+                ? () =>
+                    dfClient.startNew(
+                      "UserDataDeleteOrchestrator",
+                      makeDeleteOrchestratorId(processable.fiscalCode),
+                      processable
+                    )
+                : ProcessableUserDataDeleteAbort.is(processable)
+                ? () =>
+                    dfClient.raiseEvent(
+                      makeDeleteOrchestratorId(processable.fiscalCode),
+                      ABORT_DELETE_EVENT,
+                      {}
+                    )
+                : undefined
+            )
+          )
+          .fold(
+            _ => {
+              context.log.warn(
+                `${logPrefix}: skipping document [${JSON.stringify(
+                  processableOrNot
+                )}]`
+              );
+              return lazyOperations;
+            },
+            lazyOp => [...lazyOperations, lazyOp]
+          ),
+      [] as ReadonlyArray<Lazy<Promise<string | void>>>
     );
 
   context.log.info(
```

Comment:
instead of logging here, can we add a log in the switch for each case? (so we can track the operation type)

Reply:
should we use AI events?

Reply:
yes that would be great